lib/import.rb

```diff
@@ -238,4 +302,10 @@ def split_sequence(sequence)
       sequence.split(', ')
     end
   end
+
+  def questions_by_question_number(question_number)
+    @questions_by_question_number ||= questions.each_with_object({}){|q, hash| hash[q.question_number] = q }
```

Comment:
```suggestion
    @questions_by_question_number ||= questions.index_by(&:question_number)
```

I think this is the same and slightly neater: https://api.rubyonrails.org/classes/Enumerable.html#method-i-index_by

Reply:
The reason I went with `each_with_object` was because, to begin with at least, I was mapping `question_number` to `question_id` rather than the full `Question` object. I think you're right that for what it is now, `index_by` is probably easier to read.